media_utils/video_puppeteer.py

```diff
@@ -37,10 +37,35 @@ class VideoPuppeteer(object):
       }
     }"""
 
-    def __init__(self, marionette, url, video_selector='video'):
+    """
+        VideoPuppeteer - Wrapper to control and introspect media elements.
+
+        Inputs:
+            marionette - The marionette instance this runs in.
+            url - the URL of the page containing the video element.
+            video_selector - the selector of the element that we want to watch.
+                This is set by default to 'video', which is what most sites use,
+                but others should work.
+            interval - The polling interval that is used to check progress.
+            set_duration - When set, the polling and checking will stop at the number
+                of seconds specified. Otherwise, this will stop at the end of the video.
+            stall_wait_time - The amount of time to wait to see if a stall has cleared.
+            timeout - The amount of time to wait until the video starts.
+    """
+
+
+    def __init__(self, marionette, url, video_selector='video', interval=1,
+                 set_duration=None, stall_wait_time=None, timeout=None):
         self.marionette = marionette
         self.test_url = url
-        wait = Wait(self.marionette, timeout=30)
+        self.interval = interval
+        self.stall_wait_time = stall_wait_time
+        if timeout:
```

Comment:
This if/else can be replaced by giving timeout a default value of `30` instead of `None` in the args list.